middlewares/reqHost.go

```diff
@@ -0,0 +1,41 @@
+package middlewares
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/containous/traefik/types"
+)
+
+var reqHostKey struct{}
+
+// ReqHostMiddleware is the struct for the middleware that adds the CanonicalDomain of the requst Host into a context for later use.
+type ReqHostMiddleware struct{}
+
+// NewReqHostMiddleware is a middleware that adds the CanonicalDomain of the requst Host into a context for later use.
+func NewReqHostMiddleware() *ReqHostMiddleware {
```

Comment:
This function is not necessary, you could directly use `&ReqHostMiddleware{}` instead of creating a function for that. Like it is done for `Compress` or `AddPrefix` middlewares